osbs/repo_utils.py

```diff
@@ -69,8 +71,19 @@ def read_yaml(yaml_data, schema):
             if path.startswith('.'):
                 path = path[1:]
 
-            logger.error('validation error (%s): %s', path or 'at top level', error.message)
-        raise
+            # get all context messages without duplicates
+            error_contexts = set()
+            for context in error.context:
+                error_contexts.add(context.message)
+
+            error_message = "{}: {}\n".format(
```

Comment:
can we avoid having `\n` for single line error? (i.e. joining all intermediate errors at the end at once)